src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1940,6 +1945,34 @@ def process_declarations(self, parent, nodes, arg_list,
         for decl in walk(nodes, Fortran2003.Derived_Type_Def):
             self._process_derived_type_decln(parent, decl, visibility_map)
 
+        # INCLUDE statements are *not* part of the Fortran language and
+        # can appear anywhere. Therefore we have to do a walk to make sure we
+        # find them if they are present.
+        incl_nodes = walk(nodes, (Fortran2003.Include_Stmt,
+                                  C99Preprocessor.Cpp_Include_Stmt))
+        if incl_nodes:
+            config = Config.get()
+            # Just generate an error message for the first include we've found
+            filename = incl_nodes[0].children[0].string
+            rtxt = (f"while processing the declarations in routine "
```

Comment:
It says "routine", but parent can also be a "module" or a "program".

Reply:
I was tempted to accuse you of being picky but then I decided I agreed. That then made me realise I had quite a lot of code duplication when reporting an error so I've simplified things.